recipes/qt/6.x.x/conanfile.py

```diff
@@ -747,3 +757,16 @@ def _create_plugin(pluginname, libname, type, requires):
             self.cpp_info.components[component_name].build_modules["cmake_find_package"].append(module)
             self.cpp_info.components[component_name].build_modules["cmake_find_package_multi"].append(module)
             self.cpp_info.components[component_name].builddirs.append(os.path.join("lib", "cmake", m))
+
+        with tools.chdir(os.path.join(self.package_folder, "lib")):
+            for d1 in os.listdir():
+                if not d1.startswith("objects-"):
+                    continue
+                with tools.chdir(d1):
+                    for m in os.listdir():
+                        module = m[:m.find("_")]
+                        with tools.chdir(m):
+                            for d2 in os.listdir():
+                                obj_files = [os.path.join(os.path.abspath(os.getcwd()), d2, f) for f in os.listdir(d2)]
+                                self.cpp_info.components["qt%s" % module].exelinkflags.extend(obj_files)
+                                self.cpp_info.components["qt%s" % module].sharedlinkflags.extend(obj_files)
```

Comment:
```suggestion
        objects_dirs = glob.glob(os.path.join(self.package_folder, "lib"), "objects-*/")
        for object_dir in objects_dirs:
            for m in os.listdir(object_dir):
                submodules_dir = os.path.join(object_dir, m)
                module = m[:m.find("_")]
                for sub_dir in os.listdir(submodules_dir):
                    submodule_dir = os.path.join(submodules_dir, sub_dir)
                    obj_files = [os.path.join(submodule_dir, file) for file in os.listdir(submodule_dir)]
                    self.cpp_info.components["qt{}".format(module)].exelinkflags.extend(obj_files)
                    self.cpp_info.components["qt{}".format(module)].sharedlinkflags.extend(obj_files)
```

I tried my best, but I might be completely wrong, please check this.

Reply:
Thanks ! I just changed `glob.glob(os.path.join(self.package_folder, "lib"), "objects-*/")` into `glob.glob(os.path.join(self.package_folder, "lib", "objects-*/"))`